test/Vesting.test.ts

```diff
@@ -0,0 +1,207 @@
+                await expect(vesting.connect(investor2).withdrawTokens())
+                    .to.emit(vesting, 'Harvest').withArgs(investor2.address, '320000000000000000')
+                let afterBalanceInvestor1 = await rewardToken.balanceOf(investor1.address);
+                let afterBalanceInvestor2 = await rewardToken.balanceOf(investor2.address);
+                expect(afterBalanceInvestor1).to.be.equal(beforeBalanceInvestor1.add("160000000000000000"));
+                expect(afterBalanceInvestor2).to.be.equal(beforeBalanceInvestor2.add("320000000000000000"));
+
+                await setCurrentTime(now + 1001 + 32400); // after 100% vesting
+                await expect(vesting.connect(investor1).withdrawTokens())
+                    .to.emit(vesting, 'Harvest').withArgs(investor1.address, '840000000000000000')
+                await expect(vesting.connect(investor2).withdrawTokens())
+                    .to.emit(vesting, 'Harvest').withArgs(investor2.address, '1680000000000000000')
+                expect(await rewardToken.balanceOf(investor1.address)).to.be.equal("1000000000000000000");
+                expect(await rewardToken.balanceOf(investor2.address)).to.be.equal("2000000000000000000");
+            })
+            it('should withdraw tokens succesfully(AllocationType - Seed & Private)', async () => {
+                let blo = await ethers.provider.getBlockNumber()
+                let now = (await ethers.provider.getBlock(blo)).timestamp
+                await vesting.setInitialTimestamp(now + 30);
+
+                await expect(vesting.addInvestors([a(investor1)],
+                    [toETH("1")], 0)
+                ).to.emit(vesting, 'AddInvestors').withArgs([investor1.address],
+                    [toETH("1")], 0)
+                await expect(vesting.addInvestors([a(investor1)],
+                    [toETH("1")], 1)
+                ).to.emit(vesting, 'AddInvestors').withArgs([investor1.address],
+                    [toETH("1")], 1)
+
+                let beforeBalanceInvestor1 = await rewardToken.balanceOf(investor1.address);
+                await setCurrentTime(now + 1031); // after cliff period
+                await expect(vesting.connect(investor1).withdrawTokens())
+                    .to.emit(vesting, 'Harvest').withArgs(investor1.address, '270000000000000000')
+                let afterBalanceInvestor1 = await rewardToken.balanceOf(investor1.address);
+                expect(afterBalanceInvestor1).to.be.equal(beforeBalanceInvestor1.add("270000000000000000"));
+
+                await setCurrentTime(now + 1031 + 32400); // after 100% vesting
+                await expect(vesting.connect(investor1).withdrawTokens())
+                    .to.emit(vesting, 'Harvest').withArgs(investor1.address, '1730000000000000000')
+                expect(await rewardToken.balanceOf(investor1.address)).to.be.equal("2000000000000000000");
+            })
+        })
```

Comment:
also add tests of the unlock check part of a certain number of periods
You check money when 10% of tokens are unlocked
Also check next test cases:
1) when cliff at the moment
2) after cliff  and only initial percentage unlock
3) initial and passed 1 period
4) initial and passed 50% periods
for Seed and Privae you can write only one test and check all tes cases inner